docs/api/io/flow_control.md

```diff
@@ -2,51 +2,66 @@
+Busy wait is a method that blocks the processor for a period of time. The processor runs at full power for the duration of the wait:
 
 [![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait)](https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait/main.cpp)
 
-Notice `printf()`; you can enable this by uncommenting the line (remove the `//`). `printf()` prints to the terminal, so you can use it to get debug information.
+Notice `wait_us()` - it is part of the [Wait API](../mbed-os-api-doxy/group__platform__wait__api.html) to busy wait a given number of microseconds.
 
-<span class="tips">**Tip:** Busy wait is inefficient, and we recommend not using it. </span>
+The Wait API is a precise and ISR-safe way to create short delays (nanoseconds to microseconds). However, we recommend not busy waiting much longer than that due to inefficiency.
 
-#### Suspend thread execution
+The following few techniques are better suited for longer waits (milliseconds to seconds) which apply to our example's use case.
 
-A better alternative to busy wait is to suspend thread execution while waiting. Use:
+### Ticker, Timeout
 
- * For C++: The `ThisThread::sleep_for` API
- * For C: The `thread_sleep_for` API.
+Tickers and Timeouts are a non-blocking, interrupt-based way of creating a time interval - your code continues to execute or sleep when there is nothing to do. The difference is that Tickers are recurring whereas Timeouts are one-off.
 
-### Ticker
+Here is an example that uses a ticker object:
 
-Tickers and timers are another way of creating a time interval. These methods allow other code to run while you are waiting. During the wait period, if no threads are running, the idle thread can automatically put the system to sleep.
+[![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Ticker)](https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Ticker/main.cpp)
 
-If you don't need the precision of a high-frequency timer or ticker, we recommend that you use LowPowerTimer or LowPowerTicker instead. This allows the system to be put in deep sleep mode.
+<span class="warnings"> **Warning:** A ticker/timeout's handler (`flip` in this example), like any ISR handlers, should return quickly and not use `printf` or APIs that are not intended for ISRs.</span>
 
-Here is an example that uses a ticker object:
+If you don't need the precision of a high-frequency ticker or timeout, we recommend that you use LowPowerTicker or LowPowerTimeout instead. This allows the system to be put in deep sleep mode.
```

Comment:
```suggestion
If you don't need the precision of a high-frequency ticker or timeout, we recommend LowPowerTicker or LowPowerTimeout instead to use. This allows the system to be put in deep sleep mode.
```

Reply:
That does not read as well as what Lingkai suggested.